internal/build/phases_test.go

```diff
@@ -47,6 +48,126 @@ func testPhases(t *testing.T, when spec.G, it spec.S) {
 		h.AssertNil(t, os.RemoveAll(dockerConfigDir))
 	})
 
+	when("#Create", func() {
+		it("creates a phase and then run it", func() {
+			lifecycle := fakeLifecycle(t, false)
+			fakePhase := &fakes.FakePhase{}
+			fakePhaseFactory := fakes.NewFakePhaseFactory(fakes.WhichReturnsForNew(fakePhase))
+
+			err := lifecycle.Create(context.Background(), false, false, "test", "test", "test", "test", "test", fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			h.AssertEq(t, fakePhase.CleanupCallCount, 1)
+			h.AssertEq(t, fakePhase.RunCallCount, 1)
+		})
+
+		when("publish and clear cache", func() {
+			it("configures the phase with the expected arguments", func() {
+				verboseLifecycle := fakeLifecycle(t, true)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedRepoName := "some-repo-name"
+				expectedRunImage := "some-run-image"
+
+				err := verboseLifecycle.Create(context.Background(), true, true, expectedRunImage, "test", "test", expectedRepoName, "test", fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertEq(t, configProvider.Name(), "creator")
+				h.AssertIncludeAllExpectedPatterns(t,
+					configProvider.ContainerConfig().Cmd,
+					[]string{"-log-level", "debug"},
+					[]string{"-run-image", expectedRunImage},
+					[]string{expectedRepoName},
+					[]string{"-skip-restore"},
+				)
+			})
+
+			it("configures the phase with root", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+
+				err := lifecycle.Create(context.Background(), true, true, "test", "test", "test", "test", "test", fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertEq(t, configProvider.ContainerConfig().User, "root")
+			})
+
+			it("configures the phase with registry access", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedRepos := "some-repo-name"
+
+				err := lifecycle.Create(context.Background(), true, true, "test", "test", "test", expectedRepos, "test", fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertSliceContains(t, configProvider.ContainerConfig().Env, "CNB_REGISTRY_AUTH={}")
+			})
+
+			it("configures the phase with the expected network mode", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedNetworkMode := "some-network-mode"
+
+				err := lifecycle.Create(context.Background(), true, true, "test", "test", "test", "test", expectedNetworkMode, fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertEq(t, configProvider.HostConfig().NetworkMode, container.NetworkMode(expectedNetworkMode))
+			})
+		})
+
+		when("publish is false and clear cache is false", func() {
```

Comment:
Please correct me if I'm wrong but I think you want to test it similarly to how we test `Analyze` (there are 5 cases: "clear cache", "clear cache is false", "publish", "publish is false" and some general tests).
I agree that it's possible that `publish` will be true and `clearCache` will be false and vice versa but I wanted to reduce the number of cases that we test. I know that it's important to cover all of the scenarios but sometimes it's ok to leave some scenarios without coverage since the other scenarios are good enough. Maybe this isn't the case and it's better to split the cases.
What do you think?

Reply:
@natalieparellano , following our discussion and since it seems from the tests that there is some relation between the value of `publish` and `clearCache`, I'll split the code to 5 cases as appears in `Analyze`.

Reply:
I've just uploaded the changes.
Please notice that the general cases don't fully cover all of the cases.
For example, I don't test that the `log-level`, `run-image` and `expectedRepo` are configured as expected when `publish` and `clearCache` are true.
I think that the current coverage is enough but we can have another discussion if you think differently.